File: src/main/java/de/florianschmitt/service/BaseAdminService.java
package de.florianschmitt.service;

import java.io.Serializable;
import java.util.Optional;

public interface BaseAdminService<ENTITY extends Serializable> {
    void save(ENTITY volunteer);

    Iterable<ENTITY> findAll();

    Optional<ENTITY> findOne(long id);

    void deleteOne(long id);
}

File: src/main/kotlin/de/florianschmitt/service/BaseAdminService.kt
package de.florianschmitt.service

import java.io.Serializable
import java.util.Optional

interface BaseAdminService<ENTITY : Serializable> {
    fun save(entity: ENTITY): ENTITY
    fun findAll(): Iterable<ENTITY>
    fun findOne(id: Long): ENTITY?
    fun deleteOne(id: Long)
}
